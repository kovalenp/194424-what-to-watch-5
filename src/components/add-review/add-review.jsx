import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Header from "../header/header";
import AddNewComment from "../add-new-comment/add-new-comment.jsx";
import { movieProps } from "../../validation/propTypes";

const AddReview = ({ movie }) => {

  return (
    <section className="movie-card movie-card--full" style={{backgroundColor: movie.background_color}}>
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img
            src={movie.background_image}
            alt={movie.name}
          />
        </div>

        <h1 className="visually-hidden">WTW</h1>
        <Header />

        <div className="movie-card__poster movie-card__poster--small">
          <img
            src={movie.poster_image}
            alt={`${movie.name} poster`}
            width="218"
            height="327"
          />
        </div>
      </div>

      <AddNewComment id={movie.id}/>
    </section>
  );
};

AddReview.propTypes = {
  movie: movieProps,
  id: PropTypes.string.isRequired,
};

const MapStateToProps = (state, ownProps) => {
  const { list } = state.MOVIES;
  return {
    /* eslint-disable eqeqeq */
    movie: list.find((m) => m.id == ownProps.id),
  };
};


export default connect(MapStateToProps, null)(AddReview);
