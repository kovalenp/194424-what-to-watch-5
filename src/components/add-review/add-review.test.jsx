import React from "react";
import renderer from "react-test-renderer";
import {AddReview} from "./add-review";
import movie from "../../mocks/movie";

jest.mock(`../header/header`, () => ({
  __esModule: true,
  default: () => () => <p>MockHeader</p>,
}));

jest.mock(`../add-new-comment/add-new-comment.jsx`, () => ({
  __esModule: true,
  default: () => () => <div>AddNewCommentMock</div>,
}));

it(`AddReview test to meet B21 criteria`, () => {
  const tree = renderer
    .create(
        <AddReview
          movie={movie}
          id={`2`}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
