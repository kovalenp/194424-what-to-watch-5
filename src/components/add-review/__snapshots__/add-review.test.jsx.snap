// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview test to meet B21 criteria 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "#73B39A",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Bronson poster"
        height="327"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
        width="218"
      />
    </div>
  </div>
</section>
`;
